src/sgame/botlib/bot_nav.cpp

```diff
@@ -408,25 +422,47 @@ void G_BotAddObstacle( const glm::vec3 &mins, const glm::vec3 &maxs, qhandle_t *
 
 		tempBox.maxs[ 0 ] += offset;
 		tempBox.maxs[ 2 ] += offset;
-		
+
 		// offset mins down by agent height so obstacles placed on ledges are handled correctly
 		tempBox.mins[ 1 ] -= params->walkableHeight;
 
 		nav->cache->addBoxObstacle( tempBox.mins, tempBox.maxs, &ref );
-		*obstacleHandle = ref;
+		obstacleHandles[i] = ref;
 	}
 }
 
-void G_BotRemoveObstacle( qhandle_t obstacleHandle )
+// We do lazy load navmesh when bots are added. The downside is that this means
+// map entities are loaded before the navmesh are. This workaround does keep
+// those obstacle (such as doors and buildables) in mind until when navmesh is
+// finally loaded, or generated.
+void BotAddSavedObstacles()
 {
-	for ( int i = 0; i < numNavData; i++ )
+	for ( std::pair<int, bbox_t> obstacle : savedObstacles )
 	{
-		NavData_t *nav = &BotNavData[ i ];
-		if ( nav->cache->getObstacleCount() <= 0 )
+		G_BotAddObstacle(obstacle.second.mins, obstacle.second.maxs, obstacle.first);
+	}
+	savedObstacles.clear();
+}
+
+void G_BotRemoveObstacle( int obstacleNum )
+{
+	auto obstacle = savedObstacles.find(obstacleNum);
```

Comment:
is it safe to just run savedObstracle.erase(obstacleNum)?

Reply:
hum, actually it is, only the iterator one has a « This shall point to a valid and dereferenceable element. » condition. But on the other hand, this makes it consistent with the one below so maybe it's better